### PR TITLE
[0.10.0] Elective Integration in Schedule

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
## Features
- Integrated Elective into Schedule
  Part of FRO-94
  - Only show chosen Elective in Schedule
    Resolves FRO-87
  - Only show chosen Elective in Schedule Glance
    Resolves FRO-95
- New Elective Period design
  Part of FRO-11
  - More Indicator under Elective Period distingushes it from Subject Period
  - Added an icon to indicate Elective Period
- Migrated to SKCom v3.3.2
  Resolves FRO-105

## Fixes
- Now Line sometimes throw hydration errors
  **Hint:** Try checking the console to see if there are any errors
- Elective Period throws when Teachers view a Classroom’s Schedule
  **Hint:** Log in as a Teacher and view a Classroom’s Schedule in the Classes page
- Hide seperator dot in Elective List Item if the Elective has no Teachers
  This can be seen in the Learn Electives page for M.301
- People Chip Set throws if the `people` prop is an empty array
  This can be seen in the Learn Electives page for M.301

## Code Changes
- Extracted Elective Grid Item component from Elective Grid
- Renamed `getSubjectName` to `formatSubjectPeriodName`

## Preview
This is the current main feature PR. Preview available at https://pr.mysk.school/.

### CODE REVIEW

**Review by smartwhatt — APPROVED** [review]
LGTM!!

**Review by Sun-52 — COMMENTED** [review]
I can't test the feature because the classroom I am in doesn't have the elective period on its schedule.

File: components/home/glance/ScheduleGlance.tsx [hunk 1]
```diff
@@ -62,7 +62,11 @@ const ScheduleGlance: StylableFC<{
       className={cn(`flex flex-col gap-3 p-4 pt-3`, className)}
     >
       <motion.div
-        key={displayPeriod?.id || displayType}
+        key={
+          displayPeriod?.id
+            ? displayPeriod.id + displayPeriod.content.length
+            : displayType
+        }
```
[the remark this answers is not in the document]
Comment (SiravitPhokeed) [reply components/home/glance/ScheduleGlance.tsx:69]: React keeps track of elements using keys. When a key changes, React treats it like the old element being killed and a new element coming in. We exploit this so Framer Motion starts the entrance animation every time the ID or the content length changes.
[Read more here.](https://www.framer.com/motion/animate-presence/##exit-animations)

File: components/schedule/ElectivePeriodDetails.tsx [hunk 2]
```diff
@@ -1,158 +1,66 @@
...
+        // Full-screen Dialog. The component was made back when <dialog> wasn’t
+        // really a thing yet.
+        // Ideally we’d show the nested Full-screen Dialogs but alas.
+        `[&_.skc-fullscreen-dialog]:sm:!hidden [&_.skc-scrim]:sm:!hidden`,
       )}
-    </AnimatePresence>
+    >
+      <ul className="space-y-2 sm:!-m-3">
+        {chosenElective && (
+          <ElectiveGridItem subject={chosenElective} enrolled />
```
[the remark this answers is not in the document]
Comment (SiravitPhokeed) [reply components/schedule/ElectivePeriodDetails.tsx:57]: `enrolled` is shorthand for `enrolled={true}`.

**Comment by SiravitPhokeed** [comment]
> @Nubboyy
> It appears that certain subjects with longer name will overflow into another line and extend the schedule box.

Should be fixed now! Have another look.

**Comment by Sun-52** [comment]
> I can't test the feature because the classroom I am in doesn't have the elective period on its schedule.

@SiravitPhokeed Could you move me to another classroom?

**Review by codeycode0101 — APPROVED** [review]

**Comment by SiravitPhokeed** [comment]
> @Sun-52
> Could you move me to another classroom?

Just did! Have another look.

**Review by Sun-52 — APPROVED** [review]
LGTM

**Pull request merged**